src/main/java/com/epam/aidial/core/security/AccessService.java

```diff
@@ -1,52 +1,117 @@
 package com.epam.aidial.core.security;
 
 import com.epam.aidial.core.ProxyContext;
+import com.epam.aidial.core.data.MetadataBase;
+import com.epam.aidial.core.data.ResourceFolderMetadata;
+import com.epam.aidial.core.data.Rule;
 import com.epam.aidial.core.service.PublicationService;
 import com.epam.aidial.core.service.ShareService;
 import com.epam.aidial.core.storage.BlobStorageUtil;
 import com.epam.aidial.core.storage.ResourceDescription;
+import com.epam.aidial.core.util.ProxyUtil;
 import com.epam.aidial.core.util.UrlUtil;
-import lombok.AllArgsConstructor;
+import io.vertx.core.json.JsonObject;
+
+import java.util.List;
 
-@AllArgsConstructor
 public class AccessService {
 
     private final EncryptionService encryptionService;
     private final ShareService shareService;
     private final PublicationService publicationService;
+    private final List<Rule> adminRules;
+
+    public AccessService(EncryptionService encryptionService,
+                         ShareService shareService,
+                         PublicationService publicationService,
+                         JsonObject settings) {
+        this.encryptionService = encryptionService;
+        this.shareService = shareService;
+        this.publicationService = publicationService;
+        this.adminRules = adminRules(settings);
+    }
+
+    public boolean hasReadAccess(ResourceDescription resource, ProxyContext context) {
+        if (hasAdminAccess(context)) {
+            return true;
+        }
 
-    public boolean hasWriteAccess(String filePath, String decryptedBucket, ProxyContext context) {
-        String expectedUserBucket = BlobStorageUtil.buildUserBucket(context);
-        if (expectedUserBucket.equals(decryptedBucket)) {
+        if (isAutoShared(resource, context)) {
             return true;
         }
-        String expectedAppDataBucket = BlobStorageUtil.buildAppDataBucket(context);
-        if (expectedAppDataBucket != null && expectedAppDataBucket.equals(decryptedBucket)) {
-            return filePath.startsWith(BlobStorageUtil.APPDATA_PATTERN.formatted(UrlUtil.encodePath(context.getSourceDeployment())));
+
+        if (resource.isPublic()) {
+            return hasPublicAccess(resource, context);
         }
-        return false;
+
+        return isMyResource(resource, context) || isAppResource(resource, context)
```

Comment:
```
if (resource.isPublic()) {
             return hasPublicAccess(resource, context);
         }
return hasAdminAccess(context) || isAutoShared(resource, context) ||  isMyResource(resource, context) || isAppResource(resource, context)
                 || hasReviewAccess(resource, context) || isSharedResource(resource, context);
```

Reply:
Changed to have admin access only to public + review resources.

Reply:
It's a developer preference. I think the statement 
```
if (isAutoShared(resource, context)) {
             return true;
         }
```
could be merge in to the last one in the method body.